### ISSUE TITLE
File Not Found

### ISSUE THREAD

**Issue by UIDEVEL** [bbilly1/tubearchivist#204]
Hi,

Great docker container, however.

I get the following issue:

[2022-03-30 05:25:39,106: WARNING/ForkPoolWorker-16] UCp3yVOm6A55nx65STpm3tXQ: get metadata from es
2022-03-30 03:25:39.136645+00:00[2022-03-30 05:25:39,135: ERROR/ForkPoolWorker-16] Task download_pending[bbf34275-b60b-40bc-a0e1-b34976a6abb6] raised unexpected: FileNotFoundError('could not find video file')
2022-03-30 03:25:39.136729+00:00Traceback (most recent call last):
2022-03-30 03:25:39.136762+00:00File "/app/home/src/index/video.py", line 371, in add_player
2022-03-30 03:25:39.136790+00:00vid_path = self.build_dl_cache_path()
2022-03-30 03:25:39.136815+00:00File "/app/home/src/index/video.py", line 365, in build_dl_cache_path
2022-03-30 03:25:39.136887+00:00raise FileNotFoundError
2022-03-30 03:25:39.136932+00:00FileNotFoundError
2022-03-30 03:25:39.136959+00:002022-03-30T03:25:39.136959096Z
2022-03-30 03:25:39.136983+00:00The above exception was the direct cause of the following exception:
2022-03-30 03:25:39.137007+00:002022-03-30T03:25:39.137007685Z
2022-03-30 03:25:39.137031+00:00Traceback (most recent call last):
2022-03-30 03:25:39.137055+00:00File "/usr/local/lib/python3.10/site-packages/celery/app/trace.py", line 451, in trace_task
2022-03-30 03:25:39.137080+00:00R = retval = fun(*args, **kwargs)
2022-03-30 03:25:39.137104+00:00File "/usr/local/lib/python3.10/site-packages/celery/app/trace.py", line 734, in __protected_call__
2022-03-30 03:25:39.137129+00:00return self.run(*args, **kwargs)
2022-03-30 03:25:39.137153+00:00File "/app/home/tasks.py", line 88, in download_pending
2022-03-30 03:25:39.137189+00:00downloader.run_queue()
2022-03-30 03:25:39.137233+00:00File "/app/home/src/download/yt_dlp_handler.py", line 58, in run_queue
2022-03-30 03:25:39.137278+00:00vid_dict = index_new_video(youtube_id)
2022-03-30 03:25:39.137305+00:00File "/app/home/src/index/video.py", line 467, in index_new_video
2022-03-30 03:25:39.137330+00:00video.build_json()
2022-03-30 03:25:39.137353+00:00File "/app/home/src/index/video.py", line 304, in build_json
2022-03-30 03:25:39.137377+00:00self.add_player()
2022-03-30 03:25:39.137400+00:00File "/app/home/src/index/video.py", line 382, in add_player
2022-03-30 03:25:39.137424+00:00raise FileNotFoundError("could not find video file") from err
2022-03-30 03:25:39.137449+00:00FileNotFoundError: could not find video file
2022-03-30 03:26:53.324517+00:00[pid: 46|app: 0|req: 44/44] 10.0.0.2 () {44 vars in 1170 bytes} 

Is this a bug, or am I doing something wrong?

**Comment by bbilly1** [comment]
What's the video in question? Can you see the file in cache/download of your volume mount? How have you mounted the volumes, can you post your docker-compose.yml?

**Comment by UIDEVEL** [comment]
I have updated the container to 0.1.3. The result is I seem to have lost half of my video's. The files are stiil present in the download folder though. Just not in the database. Is there a way to add them again? I have not yet made a backup, since I have not yet all channels downloaded. I have over 300k videos in the download folder, of which over 150k have been lost from the database(webui). 
The last video downloaded states as 11 march, while I downloaded some video's with date 29th of march.

The docker-compose.yml is not present on the system, since I use "TrueNAS Scale 22.02.0" It has a visual webui where I put all the specifications of a docker container. I can try to find a way to export it, but that'll probably involve a screenshot.

**Comment by UIDEVEL** [comment]
I have seen this issues before. Seems to be related to filesystem/harddrive issues. For some reason, after a few days online my computer starts to write random gibberish to the harddrives. This happens to any computer regardless of storage medium, harddrive condition, age etc. I have had computers that had there filesystem corrupted after 3 months of operation. If you have a better idea, please let me know. I will try to post docker compose settings later.

**Comment by UIDEVEL** [comment]
when I press last page on the webui, I get "Server Error (500)"

**Comment by bbilly1** [comment]
First, it's best to group these messages together, people willing to help and are subscribed will get a lot of notifications. Discord is a better place to exchange messages like that.

There is not much an application can do if you have filesystem or hardware issues, well except backups: Before the update an automatic backup should have been created, you can restore from there. You can also rescan your filesystem, to index any missing videos that are on your filesystem. Check out the wiki for more details.

Your issue sounds like your mounts are not available, or only temporarily and then disappear again. 

Yes, there is a bug in the last page of the link builder, thanks for mentioning it, I'll look into it.

**Comment by UIDEVEL** [comment]
Ok. Sorry for that. I tried to activate the backup, but now all video's are gone from the webui (not only the last few weeks).
I started rescanning the filesystem, and it seems that the files are being added again as we speak. Could take 2 weeks though, I have a lot of video's. I will send a message once the issue is resolved. About filesystem issues/hardware issues. They should not be present. However, regardless of passing S.M.A.R.T. tests on the harddrives, I have seen in the past issues can still occur there.

I will keep you posted.
